### PR TITLE
Draft PR showcasing how to use filtered queries in the extension.

### PR DESCRIPTION
**Pull request by starcke**
This is draft/WIP example on how to use an `extensible` predicate in the extraction queries, it showcases and relies on using the queries from https://github.com/github/codeql/pull/13852.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts [hunk 1]
```diff
@@ -380,13 +385,54 @@ export class DataExtensionsEditorView extends AbstractWebview<
       let predictedModeledMethods: Record<string, ModeledMethod>;
 
       if (useLlmGenerationV2()) {
+        // Generate a qlpack with a filter that only includes the usages we want to model.
+        const packDir = (await dir({ unsafeCleanup: true })).path;
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts:389]: This dir will be cleaned when vscode exits?

File: extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts [hunk 2]
```diff
@@ -380,13 +385,54 @@ export class DataExtensionsEditorView extends AbstractWebview<
...
+
+        const syntheticConfigPack = {
+          name: "codeql/automodel-filter",
+          version: "0.0.0",
+          library: true,
+          extensionTargets: {
+            [`codeql/${this.databaseItem.language}-all`]: "*",
+          },
+          dataExtensions: ["filter.yml"],
+        };
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts:399]: Nice

File: extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts [hunk 3]
```diff
@@ -380,13 +385,54 @@ export class DataExtensionsEditorView extends AbstractWebview<
...
+        await writeFile(filterFile, dumpYaml(filter), "utf8");
+
         const usages = await runAutoModelQueries({
           mode: this.mode,
           cliServer: this.cliServer,
           queryRunner: this.queryRunner,
           queryStorageDir: this.queryStorageDir,
           databaseItem: this.databaseItem,
           progress: (update) => progress({ ...update, maxStep }),
+          extraExtensionPacks: [packDir],
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts:435]: I was expecting to see the name of the extension pack here. I haven't looked at the entire implementation, so you may already be doing this properly. The `--extension-packs` argument must be the name (and optional version) of the extension/model pack you want to use. And this pack must be available either in the package cache or from addtional-packs or from the workspace.

**Review by aeisenberg — COMMENTED** [review]
Nice. I think this is reasonable.

**Comment by starcke** [comment]
Closing in favor of https://github.com/github/vscode-codeql/pull/2670

**Pull request closed without merging**